src/magic_folder/test/common.py

```diff
@@ -365,3 +366,6 @@ class AsyncBrokenTestCase(_TestCaseMixin, TestCase):
     run_tests_with = EliotLoggedRunTest.make_factory(
         AsynchronousDeferredRunTestForBrokenTwisted.make_factory(timeout=60.0),
     )
+
+_TWISTED_TEST_CASE = _TwistedSynchronousTestCase()
+success_result_of = _TWISTED_TEST_CASE.successResultOf
```

Comment:
this smells a little weird; at least worth a comment explaining why it's required. I also don't see `_TWISTED_TEST_CASE` used anywhere else, so can it be dropped?

Reply:
> I also don't see `_TWISTED_TEST_CASE` used anywhere else, so can it be dropped?

I was vaguely thinking that `failureResultOf` and `assertNoResult` might also be useful, and would re-use the dummy object. The use case isn't as clear for those though (and isn't needed for this PR), so I didn't included them.